bittensor/_dataloader/__init__.py

```diff
@@ -72,4 +76,5 @@ def check_config( cls, config: 'bittensor.Config' ):
         assert config.dataloader.block_size > 0, 'Block size must be larger than 0'
         assert config.dataloader.max_corpus_size > 0, 'max_corpus_size must be larger than 0'
         assert config.dataloader.num_workers >= 0, 'num_workers must be equal to or larger than 0'
+        assert config.dataloader.dataset in ['genesis','wikitext','test','validation'], 'dataset must be one of the following choices: genesis, wikitext, test, or validation'
 
```

Comment:
Perhaps i misunderstood here. Not sure if test and validation should be in the same place/hierarchy as genesis and wikitext. For example, We can divide wiki into test and validation.

But what are test and validation? Are they also datasets or subsets of datasets?

Reply:
test and validation are the test and validation subsets of the wikitext dataset. I left it separate so that when we do run a test/validation run, we can be sure that it was never sampled during training. Could rename it if you want